src/lib/transition_router/initial_validator.ml

```diff
@@ -87,6 +87,7 @@ let handle_validation_error ~logger ~rejected_blocks_logger ~time_received
         "Error in verifier verifying blockchain proof for $state_hash: $error" ;
       Deferred.unit
   | `Invalid_proof ->
+      Mina_metrics.(Counter.inc_one Rejected_blocks.invalid_proof) ;
```

Comment:
Could you add a debug log here as well? with protocol state and the proof in the metadata

Reply:
We should probably add an alert for this metric. Seems like we would quickly want to triage and address this if it start happening.